core/src/main/java/com/redhat/lightblue/client/response/LightblueResponse.java

```diff
@@ -101,6 +101,11 @@ private int parseInt(String fieldName) {
         }
 
         try {
+            JsonNode errorCode = json.path("errorCode");
+            if (errorCode != null && !errorCode.isNull() && !errorCode.isMissingNode()){
+                throw new LightblueResponseParseException("Error parsing lightblue response: Error returned. JSON returned:" + json.toString() + "\n");
+            }
+
```

Comment:
I got some NPE in the tests and the exception was pretty similar something I already saw on ligthblue itself, but I confess I didn't look the implementation of has Error , it might be better if there is something missing to put on that method. I will check